dockertest/output/dockerinfo_unittests.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import types
+from unittest2 import TestCase, main
+
+# DO NOT allow this function to get loose in the wild!
+def mock(mod_path):
+    """
+    Recursively inject tree of mocked modules from entire mod_path
+    """
+    name_list = mod_path.split('.')
+    child_name = name_list.pop()
+    child_mod = sys.modules.get(mod_path, types.ModuleType(child_name))
+    if len(name_list) == 0:  # child_name is left-most basic module
+        if child_name not in sys.modules:
+            sys.modules[child_name] = child_mod
+        return sys.modules[child_name]
+    else:
+        # New or existing child becomes parent
+        recurse_path = ".".join(name_list)
+        parent_mod = mock(recurse_path)
+        if not hasattr(sys.modules[recurse_path], child_name):
+            setattr(parent_mod, child_name, child_mod)
+            # full-name also points at child module
+            sys.modules[mod_path] = child_mod
+        return sys.modules[mod_path]
+
+setattr(mock('autotest.client.shared.error'), 'AutotestError', Exception)
+setattr(mock('autotest.client.shared.error'), 'CmdError', Exception)
+setattr(mock('autotest.client.shared.error'), 'TestError', Exception)
+setattr(mock('autotest.client.shared.error'), 'TestFail', Exception)
+setattr(mock('autotest.client.shared.error'), 'TestNAError', Exception)
+setattr(mock('autotest.client.utils'), 'PlaceHolder', Exception)
+
+
+class TestDockerInfo(TestCase):
+
+    def setUp(self):
+        from dockertest.output import DockerInfo
+        self.DockerInfo = DockerInfo
+
+    def tearDown(self):
+        del self.DockerInfo
+
+    def test_basic(self):
+        info_string = """Containers: 0
+ Running: 0
+ Paused: 0
+ Stopped: 0
+Images: 3
+Server Version: 1.12.6
+Storage Driver: devicemapper
+ Pool Name: vg--docker-docker--pool
+ Pool Blocksize: 524.3 kB
+ Data file: %s
+ Library Version: 1.02.135-RHEL7 (2016-11-16)
+Logging Driver: journald
+Plugins:
+ Volume: local lvm
+ Network: host null bridge overlay
+ID: 5UCC:ANAG:4BIE:2KK6:VGP4:XKVO:5HB5:RPOA:PFLJ:HXRF:FCDV
+Insecure Registries:
+ 127.0.0.0/8""" % ''
```

Comment:
Oooohhh, okay, so it's actually testing something useful then, gotcha.  Couldn't tell if it was a typo or not (they all look the same to me) hahhah 😀